nautobot_dns_models/template_content.py

```diff
@@ -0,0 +1,39 @@
+"""Extensions of baseline Nautobot views."""
+
+from nautobot.apps.ui import TemplateExtension
+
+from .models import AAAARecordModel, ARecordModel
+
+
+class IPAddressDNSRecords(TemplateExtension):  # pylint: disable=abstract-method
+    """Add DNS A Records to the right side of the IP Address page."""
+
+    model = "ipam.ipaddress"
+
+    def right_page(self):
+        """Add content to the right side of the IP Address page."""
+        dns_records = None
+        dns_records_type = None
+
+        # Check if the IP address is IPv4 or IPv6
+        ip_version = self.context["object"].ip_version
+        if ip_version == 4:
+            dns_records = ARecordModel.objects.filter(
+                address=self.context["object"],
+            )
+            dns_records_type = "A"
+        elif ip_version == 6:
+            dns_records = AAAARecordModel.objects.filter(
+                address=self.context["object"],
+            )
+            dns_records_type = "AAAA"
```

Comment:
does it make sense to return a None here is nothing is shown, to conditionally show the record? I suspect most of the time there will not be a record

Reply:
Handling this in the HTML template. Added a screenshot to show this. Would you prefer to instead avoid showing the panel altogether if there are no records?

Reply:
That would be my thought

Reply:
I've added buttons to add records so I guess now it makes sense to keep the panel even if there are no records.

Reply:
After all, a Constance setting was added to control the desired behavior.